src/docs/development/tools/devtools/visual-debugging.md

```diff
@@ -0,0 +1,221 @@
+
+* [Flutter documentation: timeDilation property](https://api.flutter.dev/flutter/scheduler/timeDilation.html) 
+
+### Example
+
+![Screen recording showing normal animation speed]({% asset tools/devtools/debug-toggle-slow-animations-disabled.gif @path %})
+![Screen recording showing slowed animation speed]({% asset tools/devtools/debug-toggle-slow-animations-enabled.gif @path %})
+
+## Show Guidelines
+
+This option draws guidelines on your app which show render boxes, alignments,
+paddings, scroll views, clippings and spacers.
+
+This tool can be used for better understanding your layout, for instance
+finding unwanted paddings or understanding how widgets are being aligned.
+
+You can also enable this in code:
+
+```dart
+debugPaintSizeEnabled = true;
+```
+
+### Render Boxes
+
+Widgets that draw to the screen create a [render box][]; they’re the 
+building blocks of Flutter layouts. They’re shown with a bright blue border:
+
+![Screenshot of render box guidelines]({% asset tools/devtools/debug-toggle-guideline-render-box.png @path %})
+
+
+### Alignments
+
+Alignments are shown with yellow arrows. These arrows show the vertical
+and horizontal offsets of a widget relative to its parent.
+For example this button’s icon is shown as being centered by the four arrows:
+
+![Screenshot of alignment guidelines]({% asset tools/devtools/debug-toggle-guidelines-alignment.png @path %})
+
+### Paddings
+
+Paddings are shown with a semi-transparent blue background:
+
+![Screenshot of padding guidelines]({% asset tools/devtools/debug-toggle-guidelines-padding.png @path %})
+
+### Scroll views
+
+Widgets with scrolling contents such as list views are shown with green arrows:
+
+![Screenshot of scroll view guidelines]({% asset tools/devtools/debug-toggle-guidelines-scroll.png @path %})
+
+### Clipping
+
+Clippings, such as when using the [ClipRect widget][], are shown
+with a dashed pink line with a scissors icon:
+
+[ClipRect widget]: https://api.flutter.dev/flutter/widgets/ClipRect-class.html
+
+![Screenshot of clip guidelines]({% asset tools/devtools/debug-toggle-guidelines-clip.png @path %})
+
+### Spacers
+
+Spacer widgets are shown with a grey background,
+such as this SizedBox without a child:
+
+![Screenshot of spacer guidelines]({% asset tools/devtools/debug-toggle-guidelines-spacer.png @path %})
+
+## Show Baselines
+
+This option makes all baselines visible.
+Baselines are horizontal lines that are used to position text.
+
+This can be useful for checking whether text is precisely aligned vertically.
+For example, here the text baselines are slightly misaligned:
+
+![Screenshot with show baselines enabled]({% asset tools/devtools/debug-toggle-guidelines-baseline.png @path %})
+
+The [Baseline][] widget can be used to adjust baselines.
+
+[Baseline]: https://api.flutter.dev/flutter/widgets/Baseline-class.html
+
+A line will be drawn on any [render box][] which has a baseline set.
+Alphabetic baselines are shown as green and ideographic as yellow.
```

Comment:
Avoiding future tense. Also, the second sentence was not a full sentence, so joined it to the first.

```suggestion
A line is drawn on any [render box][] that has a baseline set;
alphabetic baselines are shown as green and ideographic as yellow.
```